### PR TITLE
Add new rust nightly

### PR DESCRIPTION
**Pull request by alippai**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by alippai** [comment]
@conda-forge-admin, please rerender

**Comment by alippai** [comment]
@conda-forge-admin, please rerender

**Comment by alippai** [comment]
@conda-forge-admin, please rerender

**Comment by pkgw** [comment]
OK, I *think* this is reasonable? @isuruf, should I be submitting PRs to this feedstock whenever there's a new version of Rust released?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ package:
   version: {{ version }}
 
 build:
-  number: 1
+  number: 2
```
Comment (isuruf) [inline recipe/meta.yaml:14]:
```suggestion
  number: 1
```

**Comment by pkgw** [comment]
Good to merge now?

**Pull request merged**